### PR TITLE
[PM-11000] AnonLayout Icon/Logo theme refactor

### PR DESCRIPTION
**Pull request by rr-bw**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11000

## 📔 Objective

Adds a new tailwind class, `marketing-logo`, to handle automatically styling marketing related logos based on the theme. This allows us to remove the `ThemeStateService`, which was a bit cumbersome and required a workaround to check for the specific theme if the user had set their theme to `system`.

In this PR, the `marketing-logo` class is being used for both the Bitwarden Logo and the Bitwarden Shield Icon.

Note:
- Removing the `ThemeStateService` means that Storybook will now show the correct color logo for both light and dark panels.

## 🎨  Storybook

https://62a88a6de5b807fa98886113-oxxsbpsfli.chromatic.com/?path=/story/auth-anon-layout--with-primary-content

## 📷 Screencast

https://github.com/user-attachments/assets/3552e49d-cde7-4581-9c7b-13a1a000a31b

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by danielleflinn** [comment]
Looks great! Thanks @rr-bw and @vleague2 for the collaboration on this

File: apps/desktop/desktop_native/Cargo.lock [hunk 1]
```diff
@@ -282,6 +282,12 @@ dependencies = [
  "piper",
 ]
 
+[[package]]
+name = "bytes"
+version = "1.7.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "8318a53db07bb3f8dca91a600466bdb3f2eaadeedfdbcf02e1accbad9271ba50"
+
```
Comment (vleague2) [inline apps/desktop/desktop_native/Cargo.lock:290]: ❓ Is this change supposed to be in here?

Comment (rr-bw) [reply apps/desktop/desktop_native/Cargo.lock:290]: Yes this is fine, but not part of my changes. This happened on another PR and Dani gave some context here: https://github.com/bitwarden/clients/pull/10492#pullrequestreview-2245350170

Comment (quexten) [reply apps/desktop/desktop_native/Cargo.lock]: I went ahead and merged the latest main into this PR which already has the cargo lock changes applied, so they are out of the diff. The one line in index.d.ts still remains as that has not been updated in main yet.

Comment (rr-bw) [reply apps/desktop/desktop_native/Cargo.lock]: @quexten Thanks!

**Review by quexten — APPROVED** [review]

**Comment by quexten** [comment]
Nice work! Seems the linting automation failed because the github action had networking issues. Might want to just re-run the action.

File: libs/auth/src/angular/icons/bitwarden-shield.icon.ts [hunk 2]
```diff
@@ -1,13 +1,7 @@
...
+    <path class="tw-fill-marketing-logo" d="M82.2944 69.1899V37.2898H60V93.9624C63.948 91.869 67.4812 89.5927 70.5998 87.1338C78.3962 81.0196 82.2944 75.0383 82.2944 69.1899ZM91.8491 30.9097V69.1899C91.8491 72.0477 91.2934 74.8805 90.182 77.6883C89.0706 80.4962 87.6938 82.9884 86.0516 85.1649C84.4094 87.3415 82.452 89.4598 80.1794 91.5201C77.9068 93.5803 75.8084 95.2916 73.8842 96.654C71.96 98.0164 69.9528 99.304 67.8627 100.517C65.7726 101.73 64.288 102.552 63.4088 102.984C62.5297 103.416 61.8247 103.748 61.2939 103.981C60.8958 104.18 60.4645 104.28 60 104.28C59.5355 104.28 59.1042 104.18 58.7061 103.981C58.1753 103.748 57.4703 103.416 56.5911 102.984C55.712 102.552 54.2273 101.73 52.1372 100.517C50.0471 99.304 48.04 98.0164 46.1158 96.654C44.1916 95.2916 42.0932 93.5803 39.8206 91.5201C37.548 89.4598 35.5906 87.3415 33.9484 85.1649C32.3062 82.9884 30.9294 80.4962 29.818 77.6883C28.7066 74.8805 28.1509 72.0477 28.1509 69.1899V30.9097C28.1509 30.0458 28.4661 29.2981 29.0964 28.6668C29.7267 28.0354 30.4732 27.7197 31.3358 27.7197H88.6642C89.5268 27.7197 90.2732 28.0354 90.9036 28.6668C91.5339 29.2981 91.8491 30.0458 91.8491 30.9097Z" fill="#175DDC"/>
```
Comment (vleague2) [inline libs/auth/src/angular/icons/bitwarden-shield.icon.ts:5]: This still has a hardcoded hex for `fill`, should that be gone now that we are using the class?

Comment (rr-bw) [reply libs/auth/src/angular/icons/bitwarden-shield.icon.ts]: Thanks for catching that! Fixed here: https://github.com/bitwarden/clients/pull/10549/commits/5828e327fa6bec0832d9ab9951d6b62c744b9788

File: libs/components/src/tw-theme.css [hunk 3]
```diff
@@ -173,9 +179,15 @@
   --color-text-alt2: 255 255 255;
   --color-text-code: 240 141 199;
 
+  --color-marketing-logo: 255 255 255;
+
   --tw-ring-offset-color: #002b36;
 }
 
+.theme_solarizedDark {
```
[the remark this answers is not in the document]
Comment (rr-bw) [reply libs/components/src/tw-theme.css]: For sure, thanks - https://github.com/bitwarden/clients/pull/10549/commits/9233901beb091875d79b94abbe9f5ce0a0f874a4

**Review by vleague2 — APPROVED** [review]

**Review by quexten — APPROVED** [review]

**Pull request merged**